### ISSUE TITLE
Purchases: renewal caused fatal error

### ISSUE THREAD

**Issue by sirbrillig** [Automattic/wp-calypso#12969]
#### Steps to reproduce
1. Starting at URL: https://wordpress.com/checkout/qingbingcha.org (I am on Staging, if it matters)
2. I clicked "Pay with credits and renew" (or whatever the button reads when you have credits)
3. The button text changed to "Completing your purchase"
4. Nothing changed for a while so I opened the JavaScript console and found this error: 

```
Uncaught TypeError: Cannot read property 'moment' of undefined
    at Object.handleCheckoutCompleteRedirect (/calypso/upgrades.6c060c5e7f6cf34bd897.m.js:23)
    at /calypso/upgrades.6c060c5e7f6cf34bd897.m.js:5
    at /calypso/build.97c89e6abb6a5d53ae8d.m.js:58
```

#### Browser / OS version

Mac OS 10.12.4, Chrome Version 57.0.2987.133 (64-bit)

#### Screenshot / Video

<img width="1439" alt="screen shot 2017-04-10 at 3 44 57 pm" src="https://cloud.githubusercontent.com/assets/2036909/24879699/b1ca383a-1e05-11e7-8f8a-c90299864f05.png">

#### Context / Source

I believe this is the line where the error occurred: https://github.com/Automattic/wp-calypso/blob/ac256d033a58e3fd10124c83f6bfd60a6294e214/client/my-sites/upgrades/checkout/checkout.jsx#L275

**Comment by designsimply** [comment]
I tested a renewal for a domain using free credits in production and was unable to reproduce the problem using the following testing steps:

1. Find any upgrade that is ready for renewal.
1. Go to Network Admin > Store Admin and make sure the upgrade owner has enough free credits to renew.
1. Log in to https://wordpress.com/ as the user.
1. Go to the renewal reminder email and click the "Renew" button.
1. Open the developer console to look for pertinent JavaScript errors.
1. Click the red "Renew Now" button.
1. Click the "Purchase $26 subscription with Credits" button.
1. Check to see if the renewal takes too long or if an error appears in the console.

<strong>Result:</strong> no related errors seemed to appear in the console and I saw a message that said the following about 12 seconds after clicking the Purchase button:

> Domain Registration has been renewed and will now auto renew in the future. Learn more

![screen shot 2017-04-11 at tue apr 11 12 51 15 pm](https://cloud.githubusercontent.com/assets/1119271/24925470/98437f24-1eb5-11e7-9441-af4153a15527.png)
<sup>Seen at https://wordpress.com/me/purchases/madefortesting.wordpress.com/4476069 using Chrome 57.0.2987.133 on Mac OS X 10.12.4 logged in as user user849 in production.</sup>

**Comment by designsimply** [comment]
@sirbrillig I checked the renewal status for WordPress.com Premium on the account using qingbingcha.org and it does look like the renewal went through around the same time you reported this issue. Did you have to click something again or were you forced to refresh the page and the renewal was just updated after that?

**Comment by sirbrillig** [comment]
@designsimply I think the renewal went through and the error was just when it tried to tell me about it.

I figured that this might not be reproducible. Possibly something about staging and/or having updated my credit total while the checkout page was open. I just figured I would create an issue while I had it open so that perhaps we could refer to it if anyone ever saw such a thing happen again. Feel free to close.
